pallets/did/src/lib.rs

```diff
@@ -184,6 +186,12 @@ pub enum StorageError {
 	/// The given DID does not contain the right key to verify the signature of
 	/// a DID operation.
 	DidKeyNotPresent(DidVerificationKeyType),
+	/// One or more verification keys referenced are not stored in the set of
+	/// verification keys.
+	VerificationKeyNotPresent(PublicVerificationKey),
+	/// Duplicate key specified in the vector of keys to delete from the
+	/// verification keys
+	DuplicateVerificationKey(PublicVerificationKey),
```

Comment:
Since I did not get a `BTreeMap` to work properly as input parameter inside an update operation, I had to switch to lists. And so we need another error in case the user is trying to delete the same key twice in the same operation.